src/backend/marsha/core/models/file.py

```diff
@@ -140,4 +153,15 @@ def get_source_s3_key(self, stamp=None):
 
         """
         stamp = stamp or to_timestamp(self.uploaded_on)
-        return "{pk!s}/document/{pk!s}/{stamp:s}".format(pk=self.pk, stamp=stamp)
+
+        # We don't want to deal with None value so we set it with an empty string
+        extension = extension or ""
+
+        # We check if extension start with a leading dot or not. If it's not the case we add
+        # it at the beginning of the string
+        if extension and extension[:1] != ".":
+            extension = "." + extension
+
+        return "{pk!s}/document/{pk!s}/{stamp:s}{extension:s}".format(
+            pk=self.pk, stamp=stamp, extension=extension
```

Comment:
I can't use `lstrip` in fact because if the extension is `None` I will have an empty string and the filename will end with a dot and this is not what we want.